### PR TITLE
[DOCS-5448] Bryce/azure programmatic guide

### PR DESCRIPTION
**Pull request by buraizu**
### What does this PR do?
Creates:
- a guide for Datadog-Azure integration programmatic management 
- a guide for Datadog-Azure native integration programmatic management

### Motivation
DOCS-4695

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/integrations/guide/azure-native-programmatic-management.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+- link: "https://docs.datadoghq.com/integrations/guide/azure-portal"
+  tag: "Documentation"
+  text: "Managing the Azure Native Integration"
+---
+
+## Overview
+
+{{% site-region region="us3" %}}
+
+The Azure Native integration uses the Datadog resource to streamline management and data collection for your Azure environment. Datadog recommends using this method when possible. This is possible through creation of the [azurerm_datadog_monitor][3] resource and assignment of the [Monitoring Reader role][4] in Azure to link your Azure subscription(s) to your Datadog organization. This replaces the App Registration credential process for metric collection and Event Hub setup for log forwarding.
```
Comment (rtrieu) [inline content/en/integrations/guide/azure-native-programmatic-management.md:18]:
```suggestion
The Azure Native integration uses the Datadog resource to streamline management and data collection for your Azure environment. Datadog recommends using this method when possible. This method involves creating the [azurerm_datadog_monitor][3] resource and assigning the [Monitoring Reader role][4] in Azure to link your Azure subscription(s) to your Datadog organization. This replaces the App Registration credential process for metric collection and Event Hub setup for log forwarding.
```

File: content/en/integrations/guide/azure-native-programmatic-management.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+[2]: /integrations/guide/azure-portal/
+[3]: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/datadog_monitors
+[4]: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/datadog_monitors#role-assignment
+[5]: https://learn.microsoft.com/en-us/azure/partner-solutions/datadog/create#configure-metrics-and-logs
+{{% /site-region %}}
+
+{{% site-region region="us,us5,eu,ap1,gov" %}}
+
+The Azure Native integration is only supported for users of Datadog's US3 site. If you're using a different [Datadog site][3], see the standard [Azure Programmatic Management guide][1]. If you're using the Datadog US3 site, [change the site selector][2] on the right of this page.
```
Comment (rtrieu) [inline content/en/integrations/guide/azure-native-programmatic-management.md:84]:
```suggestion
The Azure Native integration is only supported for users of Datadog's US3 site. If you're using a different [Datadog site][1], see the standard [Azure Programmatic Management guide][2]. If you're using the Datadog US3 site, [change the site selector][3] on the right of this page.
```

File: content/en/integrations/guide/azure-native-programmatic-management.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+[1]: /integrations/guide/azure-programmatic-management/
```
Comment (rtrieu) [inline content/en/integrations/guide/azure-native-programmatic-management.md:86]:
```suggestion
[1]: /getting_started/site/
```

File: content/en/integrations/guide/azure-native-programmatic-management.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+[2]: ?site=us3
```
Comment (rtrieu) [inline content/en/integrations/guide/azure-native-programmatic-management.md:87]:
```suggestion
[2]: /integrations/guide/azure-programmatic-management/
```

File: content/en/integrations/guide/azure-native-programmatic-management.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+[3]: /getting_started/site/
```
Comment (rtrieu) [inline content/en/integrations/guide/azure-native-programmatic-management.md:88]:
```suggestion
[3]: ?site=us3
```

File: content/en/integrations/guide/azure-native-programmatic-management.md (hunk 2 above)
Comment (rtrieu) [reply content/en/integrations/guide/azure-native-programmatic-management.md:84]: I don't think this impacts the functionality of the links, but figured it wouldn't hurt to make them consecutive :)

File: content/en/integrations/guide/azure-programmatic-management.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+## Overview
+
+This guide demonstrates how to programmatically manage the Azure integration with Datadog, as well as other Azure resources such as the Datadog Agent VM extension. This enables you to manage observability across multiple accounts at once.
+
+## Datadog Azure integration
+
+### Azure or Azure Native integration?
+
+All sites: Users of all [Datadog Sites][3] can use the app registration credential process to implement the standard Azure integration with Datadog.
```
Comment (rtrieu) [inline content/en/integrations/guide/azure-programmatic-management.md:19]:
```suggestion
**All sites**: Users of all [Datadog Sites][3] can use the app registration credential process to implement the standard Azure integration with Datadog.
```

File: content/en/integrations/guide/azure-programmatic-management.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+US3: If your org is on the Datadog US3 site, you can use the Azure Native integration to streamline management and data collection for your Azure environment. Datadog recommends using this method when possible. Setup entails creating a [Datadog resource in Azure][14] to link your Azure subscriptions to your Datadog organization. This replaces the app registration credential process for metric collection and Event Hub setup for log forwarding.
```
Comment (rtrieu) [inline content/en/integrations/guide/azure-programmatic-management.md:21]:
```suggestion
**US3**: If your org is on the Datadog US3 site, you can use the Azure Native integration to streamline management and data collection for your Azure environment. Datadog recommends using this method when possible. Setup entails creating a [Datadog resource in Azure][14] to link your Azure subscriptions to your Datadog organization. This replaces the app registration credential process for metric collection and Event Hub setup for log forwarding.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Sorry this review took so long. A few small suggestions, but it otherwise looks like a great update!

**Comment by buraizu** [comment]
@rtrieu thank you for the review, I've applied your suggestions.

**Comment by buraizu** [comment]
Received approval from Azure team via Slack.

**Pull request merged**